oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -502,7 +502,7 @@ private void configureEstimators(IndexingProgressReporter progressReporter) {
     private long getEstimatedDocumentCount() {
         MongoConnection mongoConnection = indexHelper.getService(MongoConnection.class);
         if (mongoConnection != null) {
-            return mongoConnection.getDatabase().getCollection("nodes").count();
+            return mongoConnection.getDatabase().getCollection("nodes").countDocuments();
```

Comment:
please use https://mongodb.github.io/mongo-java-driver/5.5/apidocs/driver-sync/com/mongodb/client/MongoCollection.html#estimatedDocumentCount(). `countDocuments` is pretty slow [1].

[1] https://mongodb.github.io/mongo-java-driver/5.5/apidocs/driver-sync/com/mongodb/client/MongoCollection.html#countDocuments()

Reply:
You are right, applied, thanks!